internal/xds/translator/cluster.go

```diff
@@ -276,10 +276,14 @@ func buildXdsClusterCircuitBreaker(circuitBreaker *ir.CircuitBreaker) *clusterv3
 	// related to pod restarts
 	cbt := &clusterv3.CircuitBreakers_Thresholds{
 		Priority: corev3.RoutingPriority_DEFAULT,
+		// concurrent retries limit = max(100, current-active-requests)
 		RetryBudget: &clusterv3.CircuitBreakers_Thresholds_RetryBudget{
 			BudgetPercent: &xdstype.Percent{
```

Comment:
if `min_rety_concurrency` is adequate, should be rm `retryBudget:100%` setting ?

Reply:
also can we disable `min_rety_concurrency` by setting it to `0` ?

Reply:
what im trying to get at is can we do without this `RetryBudget` knob, if the 20% default is too low, can we increase `maxParallelRequests` (guesing the retry budget percentage is off this value) in the circuit breaker settings ?

Reply:
I understand them as being complementary, with min retries acting as a guardrail. The budget-based limit will dynamically scale with traffic. However, in smaller numbers, the budget could be a problem. Imagine that you had some traffic that generated transient failures, and then traffic drops to 0: this means that the budget-based limit is now also 0 and all retries are aborted.

Reply:
@guydc isnt that what the knob is designed for ? to not invest more cycles when failure is evident ?
since youve already added the host predicate setting for retry it should hopefully retry and achieve success

Reply:
> what im trying to get at is can we do without this `RetryBudget` knob, if the 20% default is too low, can we increase `maxParallelRequests` (guesing the retry budget percentage is off this value) in the circuit breaker settings ?

The circuit breaker value only define the upper limit of the budget. But, the actual limit is calculated dynamically from the volume of traffic seen. So, increasing one of the circuit breakers wouldn't necessarily produce the expected outcome (actual retry limit being increased).

Reply:
based on issues like https://github.com/envoyproxy/envoy/issues/30205, id like to avoid this setting for now

Reply:
@arkodg - Sure, that makes sense. During the API discussion for retries, we preferred the more recent budget-based approach. 

I have to admit that I'm also a bit surprised by the behavior of this feature, and this follow-up PR is trying to further tweak our defaults so that our upgrades are hitless. Maybe the best approach would be to wait for this feature to receive some additional improvements, like the ones outlined in the mentioned issue. 

In that case, I can remove the budget altogether and replace it with some default min_retry value that's derived from one of the circuit breakers (e.g. 20% of parallel requests) . WDYT? Is that ok?